components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests

```diff
@@ -141,21 +144,58 @@ function install_kfp() {
   echo "[Installing KFP] Pipeline pods are ready"
 }
 
-function generate_iam_role_name() {
+function generate_oidc_role_name() {
   OIDC_ROLE_NAME="$(echo "${DEPLOY_NAME}-kubeflow-role" | cut -c1-64)"
-  OIDC_ROLE_ARN="arn:aws:iam::$(aws sts get-caller-identity --query=Account --output=text):role/${OIDC_ROLE_NAME}"
+  OIDC_ROLE_ARN="arn:aws:iam::${AWS_ACCOUNT_ID}:role/${OIDC_ROLE_NAME}"
 }
 
-function install_generated_role() {
+function install_oidc_role() {
   kubectl patch serviceaccount -n ${KFP_NAMESPACE} ${KFP_SERVICE_ACCOUNT} --patch '{"metadata": {"annotations": {"eks.amazonaws.com/role-arn": "'"${OIDC_ROLE_ARN}"'"}}}'
 }
 
-function delete_generated_role() {
+function delete_oidc_role() {
   # Delete the role associated with the cluster thats being deleted
   aws iam detach-role-policy --role-name "${OIDC_ROLE_NAME}" --policy-arn arn:aws:iam::aws:policy/AmazonSageMakerFullAccess
+  aws iam delete-role-policy --role-name "${OIDC_ROLE_NAME}" --policy-name AllowAssumeRole
   aws iam delete-role --role-name "${OIDC_ROLE_NAME}"
 }
 
+function generate_assumed_role() {
+  # If not defined in the env file
+  if [[ -z "${ASSUMED_ROLE_NAME}" ]]; then
+    ASSUMED_ROLE_NAME="${DEPLOY_NAME}-assumed-role"
+
+    # Create a trust file that allows the OIDC role to authenticate
+    local assumed_trust_file="assumed-role-trust.json"
+    printf '{
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Principal": {
+            "AWS": "arn:aws:iam::'"${AWS_ACCOUNT_ID}"':root"
+          },
+          "Action": "sts:AssumeRole",
+          "Condition": {}
+        }
+      ]
+    }' > "${assumed_trust_file}"
+    aws iam create-role --role-name "${ASSUMED_ROLE_NAME}" --assume-role-policy-document file://${assumed_trust_file} --output=text --query "Role.Arn"
+    aws iam attach-role-policy --role-name ${ASSUMED_ROLE_NAME} --policy-arn arn:aws:iam::aws:policy/AmazonSageMakerFullAccess
+  fi
+
+  # Generate the ARN using the role name
+  ASSUMED_ROLE_ARN="arn:aws:iam::${AWS_ACCOUNT_ID}:role/${ASSUMED_ROLE_NAME}"
```

Comment:
why are we generating an assumed role for every test ?

Reply:
Otherwise we would need to provide instructions to others who want to run these tests about how to create an IAM role that works with the OIDC role. This way, the entire set up of the IAM structure is done through the `run_integration_tests` script.

Reply:
Oh, now I see what Suraj meant earlier. Resolving this.